tacitDB.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import "github.com/jinzhu/gorm"
+
+type tacitDB interface {
+	autoMigrate(values ...interface{})
+	create(value interface{}) tacitDB
+	first(out interface{}, where ...interface{})
```

Comment:
Can this result in a record not found?

Reply:
It can. I added errors for login and createUser, as well as further testing, that use RecordNotFound functionality. These changes are fairly minor and can be removed easily, but they stretch across our user and DB files (and their tests). This is something I would like to discuss in our next meeting.